### ISSUE TITLE
More intuitive treatment of `type_*` and `cat_*` mappings

### ISSUE THREAD

**Issue by danielhuppmann** [iiasa/message_ix#21]
Currently, mappings between sets and the category-mapping sets have to be added explicitly. This should be made "more automatic", so that a bound on a set element (e.g., an emission) can be added without the need to define the category mapping (to itself) explicitly.

**Comment by gidden** [comment]
Agreed. `add_spatial_sets` goes some way to alleviate this on the spatial side. I would suggest we add a similar function for emissions. Are there other cases as well?
